citus-enterprise.spec

```diff
@@ -44,6 +45,201 @@ make %{?_smp_mflags}
+
+In these cases it is assumed the operator has set up appropriate security
+themselves.
+
+So, with the default settings Citus clusters are not safe from
+Man-In-The-Middle attacks. To secure the traffic completely you need to follow
+the practices outlined here:
+https://citus.com/some-page-on-secure-setup
+
+Please confirm that you have read this and understand that you should set up
+TLS yourself to send traffic between nodes securely:
+y/N?"
+
+CITUS_ACCEPT_ENCRYPTION_WARNING="\${CITUS_ACCEPT_ENCRYPTION_WARNING:-}"
+while [ -z "\$CITUS_ACCEPT_ENCRYPTION_WARNING" ]; do
+    echo "\$ssl_warning_text"
+    read -r CITUS_ACCEPT_ENCRYPTION_WARNING
+done
+
+case "\$CITUS_ACCEPT_ENCRYPTION_WARNING" in
+    y|Y|Yes|YES|yes );;
+    * )
+        echo "ERROR: Warning about encrypted traffic must be accepted before installing"
+        exit 1
+esac
+
+# create a temporary directory for gpg to use so it doesn't output warnings
+temp_gnupghome="\$(mktemp -d)"
+CITUS_LICENSE_KEY="\${CITUS_LICENSE_KEY:-}"
+while [ -z "\$CITUS_LICENSE_KEY" ]; do
+    echo ''
+    echo 'Please enter license key:'
+    read -r CITUS_LICENSE_KEY
+done
+
+# Try to decrypt the first file in the list to check if the key is correct
+if ! gpg --output "/dev/null" \
+        --batch --no-tty --yes --quiet \
+        --passphrase "\$CITUS_LICENSE_KEY" \
+        --homedir "\$temp_gnupghome" \
+        --decrypt "\$(head -n 1 "\$secret_files_list").gpg" 2> /dev/null; then
+    echo "ERROR: Invalid license key supplied"
+    exit 1
+fi
+
+echo "License key is valid"
+echo "Installing..."
+
+# Decrypt all the encrypted files
+while read -r path_unencrypted; do
+    path_encrypted="\$path_unencrypted.gpg"
+    # decrypt the encrypted file
+    gpg --output "\$path_unencrypted" \
+        --batch --no-tty --yes --quiet \
+        --passphrase "\$CITUS_LICENSE_KEY" \
+        --homedir "\$temp_gnupghome" \
+        --decrypt "\$path_encrypted"
+
+    # restore permissions and ownership
+    chmod --reference "\$path_encrypted" "\$path_unencrypted"
+    chown --reference "\$path_encrypted" "\$path_unencrypted"
+done < "\$secret_files_list"
+
+# remove the temporary gpg directory
+rm -rf "\$temp_gnupghome"
+EOF
+
+#------- END OF DECRYPT SCRIPT --------
+
+chmod +x "$bindir/citus-enterprise-pg-%{pgmajorversion}-setup"
+
+cat "$bindir/citus-enterprise-pg-%{pgmajorversion}-setup"
+
+
+
+%post
+installation_message="
++--------------------------------------------------------------+
+Please run 'sudo citus-enterprise-pg-%{pgmajorversion}-setup'
+to complete the setup of Citus Enterprise
++--------------------------------------------------------------+
+"
+echo "$installation_message"
+
+
+%preun
+libdir="%{pginstdir}/lib"
+
+secret_files_list="$libdir/citus_secret_files.metadata"
+
+# Cleanup all de decrypted files since these are not managed by the package
+# manager and would be left around otherwise
+while read -r path_unencrypted; do
+    rm -f "$path_unencrypted"
```

Comment:
Info: the `rm` here causes a warning during package builds, but that is not really harmful in any way
> W: dangerous-command-in-%preun rm